src/Filesystem/Folder.php

```diff
@@ -194,16 +194,32 @@ public function read($sort = true, $exceptions = false, $fullPath = false)
             if ($fullPath) {
                 $name = $item->getPathname();
             }
+
             if ($item->isDir()) {
-                $dirs[] = $name;
+                $dirs[$item->getCTime()][] = $name;
             } else {
-                $files[] = $name;
+                $files[$item->getCTime()][] = $name;
```

Comment:
I don't think applying `sort` function against 2d array works as you expected. For example Can your commit keep the same order if their are 3 files like the following:
```
ctime   name
100     a.txt
100     c.txt
101     b.txt
```